src/modules/registration/components/Registration.js

```diff
@@ -1,43 +1,146 @@
-import React from 'react'
+import React, { useState } from 'react'
 import styles from '../styles/Registration.module.css'
+import axios from 'axios'
+import { LoadingSpinner } from '../../common/components/LoadingSpinner'
 
 export const Registration = () => {
+    const [firstName, setFirstName] = useState('')
+    const [lastName, setLastName] = useState('')
+    const [username, setUsername] = useState('')
+    const [email, setEmail] = useState('')
+    const [password, setPassword] = useState('')
+    const [error, setError] = useState(null)
+    const [isLoading, setIsLoading] = useState(false)
+    const [success, setSuccess] = useState(false)
+
+    const handleFirstName = (event) => {
+        setFirstName(event.target.value)
+    }
+
+    const handleLastName = (event) => {
+        setLastName(event.target.value)
+    }
+
+    const handleUsername = (event) => {
+        setUsername(event.target.value)
+    }
+
+    const handleEmail = (event) => {
+        setEmail(event.target.value)
+    }
+
+    const handlePassword = (event) => {
+        setPassword(event.target.value)
+    }
+
+    const handleSubmit = async (event) => {
+        event.preventDefault()
+        setIsLoading(true)
+        try {
+            const response = await axios.post(
+                'https://cf4lq5bf3denzvriyvqnuxlnxu0gwtgo.lambda-url.ap-southeast-2.on.aws/',
+                {
+                    firstName,
+                    lastName,
+                    username,
+                    email,
+                    password,
+                },
+                {
+                    headers: {
+                        'Access-Control-Allow-Origin': '*',
+                    },
+                    mode: 'cors',
+                }
+            )
+            console.log(response.data)
+            setSuccess(true)
+        } catch (error) {
+            console.log('Error:', error)
+            setError(error)
```

Comment:
Unless we want to display the actual message from the server we wouldnt set the whole object here. 
Lets just set it to 
` setError(true)` as thats all we are really doing here. 
Can you then also update naming to 
`isError` instead of `error`

Reply:
done